### ISSUE TITLE
400 Bad Request when syncing filters

### ISSUE THREAD

**Issue by fouram** [bakito/adguardhome-sync#97]
**Describe the bug**
400 Bad Request when syncing filters

**To Reproduce**
1. What version of AdGuardHome sync used? 0.4.6 (2022-09-07T20:13:05Z)
2. What version of AdGuardHome us used? v0.107.12 (also occurred on v0.107.11
3. How does the configuration look?
```
# cron expression to run in daemon mode. (default; "" = runs only once)
cron: "*/10 * * * *"

# runs the synchronisation on startup
runOnStart: true

origin:
  # url of the origin instance
  url: https://10.0.20.8
  # apiPath: define an api path if other than "/control"
  # insecureSkipVerify: true # disable tls check
  username: [REDACTED]
  password: [REDACTED]

# replica instance (optional, if only one)
replica:
  # url of the replica instance
  url: https://10.0.20.9
  username: [REDACTED]
  password: [REDACTED]

# replicas instances (optional, if more than one)
#replicas:
  # url of the replica instance
#  - url: http://192.168.1.3
#    username: username
#    password: password
#  - url: http://192.168.1.4
#    username: username
#    password: password
    # autoSetup: true # if true, AdGuardHome is automatically initialized. 

# Configure the sync API server, disabled if api port is 0
api:
  # Port, default 8080
  port: 8080
  # if username and password are defined, basic auth is applied to the sync API 
 # username: username
 # password: password

# Configure sync features; by default all features are enabled.
features:
  generalSettings: true
  queryLogConfig: true
  statsConfig: true
  clientSettings: true
  services: true
  filters: true
  dhcp:
    serverConfig: true
    staticLeases: true
  dns:
    serverConfig: true
    accessLists: true
    rewrites: true
```
5. What is the error message? `ERROR   sync    sync/sync.go:236        Error syncing filters   {"from": "10.0.20.8", "to": "10.0.20.9", "error": "400 Bad Request"}`

**Expected behavior**
This is the initial sync, I would expect the sync to complete successfully.

**Log Files**
```
2022-09-12T04:45:47.360Z        INFO    root    cmd/root.go:118 Using config file:/opt/adguardhome-sync/adguardhome-sync.yaml
2022-09-12T04:45:47.363Z        INFO    sync    sync/sync.go:31 AdGuardHome sync        {"version": "0.4.6", "build": "2022-09-07T20:13:05Z"}
2022-09-12T04:45:47.364Z        INFO    sync    sync/sync.go:57 Setup cronjob   {"cron": "*/10 * * * *", "next-execution": "2022-09-12T04:50:00.000Z"}
2022-09-12T04:45:47.364Z        INFO    sync    sync/http.go:52 Starting API server     {"port": 8080}
2022-09-12T04:45:47.365Z        INFO    sync    sync/sync.go:67 Running sync on startup
2022-09-12T04:45:47.471Z        INFO    sync    sync/sync.go:115        Connected to origin     {"from": "10.0.20.8", "version": "v0.107.12"}
2022-09-12T04:45:47.554Z        INFO    sync    sync/sync.go:198        Start sync      {"from": "10.0.20.8", "to": "10.0.20.9"}
2022-09-12T04:45:47.574Z        INFO    sync    sync/sync.go:206        Connected to replica    {"from": "10.0.20.8", "to": "10.0.20.9", "version": "v0.107.12"}
2022-09-12T04:45:47.616Z        INFO    client  client/client.go:248    Add filter      {"host": "10.0.20.9", "url": "https://github.com/AdAway/adaway.github.io/blob/master/hosts.txt", "whitelist": false, "enabled": true}
2022-09-12T04:45:48.296Z        ERROR   sync    sync/sync.go:236        Error syncing filters   {"from": "10.0.20.8", "to": "10.0.20.9", "error": "400 Bad Request"}
github.com/bakito/adguardhome-sync/pkg/sync.(*worker).syncTo
        /home/bakito/go/src/github.com/bakito/adguardhome-sync/pkg/sync/sync.go:236
github.com/bakito/adguardhome-sync/pkg/sync.(*worker).sync
        /home/bakito/go/src/github.com/bakito/adguardhome-sync/pkg/sync/sync.go:186
github.com/bakito/adguardhome-sync/pkg/sync.Sync.func3
        /home/bakito/go/src/github.com/bakito/adguardhome-sync/pkg/sync/sync.go:68
```

Additional context:
TLS is configured for both instances and the servers (as well as client machines) trust the local CA certificate. Configuration was tested with `insecureSkipVerify: true` but the result is the same

**Comment by fouram** [comment]
Duplicate of https://github.com/bakito/adguardhome-sync/issues/80 and others.

I'm leaving this open for now as perhaps this is an indication that logging/messaging of this error/situation could be better? Even if there was mention of it in the documentation. It also doesn't help that all the previous instances of this being reported were closed with seemingly no changes to logging or documentation? Unless I missed something (quite possible, I was up quite late)

**Comment by bakito** [comment]
@fouram I modified how the error message is presented by adding the replica response body.

`2022-09-12T18:40:01.685+0200    ERROR   sync    sync/sync.go:236        Error syncing filters   {"from": "ha.bakito.net:3000", "to": "127.0.0.1:9091", "error": "400 Bad Request(Couldn't fetch filter from url https://github.com/AdAway/adaway.github.io/blob/master/hosts.txt: data is HTML, not plain text\n)"}
`

**Comment by fouram** [comment]
Nice. I suppose given the fact that it could be a bad URL for a number of reasons (list extinct, incorrect github page, etc) this is the best solution. Thank you!

**Comment by bakito** [comment]
fixed in https://github.com/bakito/adguardhome-sync/releases/tag/v0.4.7
